packages/common/openapi.yml

```diff
@@ -0,0 +1,132 @@
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/OpportunitySummary"
+                  
+    post:
+      requestBody:
+        description: Create a new opportunity
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/Opportunity"
+      responses:
+        "200":
+          description: ok
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Opportunity"
+        
+  /opportunity/{opportunityId}:
+    get:
+      parameters:
+      - name: opportunityId
+        in: path
+        required: true
+        schema:
+          type: string
+      responses:
+        '200':
+          description: Opportunity
+          content:
+            application/json:
+              schema: 
+                $ref: "#/components/schemas/Opportunity"
+
+    put:
+      parameters:
+      - name: opportunityId
+        in: path
+        required: true
+        schema:
+          type: string 
+      requestBody:
+        description: Update an existing opportunity
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/Opportunity"
+      responses:
+        '200':
+          description: Opportunity
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Opportunity"
+
+  /user:
+    get:
+      responses:
+        '200':
+          description: User
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/User"
+  
+components:
+  schemas:
+    OpportunitySummary:
+      type: object
+      properties:
+        opportunityId:
+          type: string
+        title:
+          type: string
+        contactName:
+          type: string
+        requiredPeopleCount:
+            type: number
+        startTime:
+          type: number
+          example: 1666817899
+        endTime:
+          type: number
+          example: 1666817900
+        postedTime:
+          type: number
+          example: 1666817898
+     
+    Opportunity:
+      type: object
+      allOf:
+      - $ref: "#/components/schemas/OpportunitySummary"
+      - properties:
+          description:
+            type: string
+          locationName:
+            type: object
+          indoorsOrOutdoors:
+            type: string
+            enum: [indoors, outdoors]
+          contactEmail:
+            type: string
+          contactPhone:
+            type: string
+          postedByUserId:
+            type: string
+          criminalRecordCheckRequired:
+            type: boolean
+          idealVolunteer:
+            type: string
+          additionalInformation:
+            type: string
+
+    User:
+      type: object
+      properties:
+        userId:
+          type: string
```

Comment:
We could probably get the name in as part of the stub at this point, we can worry about roles later.